cmd/spx/template/runner.html

```diff
@@ -44,19 +44,54 @@
 				document.getElementById('tab-loader').style.display = 'none';
 			}
 		}
-		window.startWithZipBuffer = async (buffer) => {
-			buffer = new Uint8Array(buffer);
+		let gameApp = null
+
+		window.startWithZipBuffer = async (buffer, isEditor = false, addInfos = null, deleteInfos = null, updateInfos = null) => {
```

Comment:
What is the expected behavior of Builder project-editor?

`startWithZipBuffer(projectData, true)` or `startWithZipBuffer(projectData, true, addInfo, deleteInfos, updateInfos)`?

Reply:
The current parameters remain consistent with before. You can pass startWithZipBuffer(projectData) to run the game directly (without caching).

However, if you want to enable caching mode, it's recommended to use the following interface. The projectName should be unique for each project:

```js
// load project at backgound 
window.startProject(buffer, isEditor = false, projectName = "Game")
// update project asset
window.updateProject(buffer, addInfos, deleteInfos, updateInfos)
// start a game base on the current project asset
window.runGame()
// stop the game
window.stopGame() 
```

Reply:
Since updating resources takes time, will the following code cause issues?

```ts
window.startProject(...)
// call `updateProject` while `startProject` is not finished
window.updateProject(...)
// or another call to `updateProject` while the first is not finished
window.updateProject(...)
```

Additionally, is it acceptable to call `runGame` immediately after `startProject`? Since `startProject` returns immediately, there's no way to wait for it to finish.

Reply:
Certainly, you can do that. In my next iteration, I'll add a check to see if there's already a request to start the game once the project has loaded. If there is a request, it will immediately run the game

Reply:
As discussed:

1. Builder do not need to wait the previous operation to finish before perform the next operation, runner will handle that
2. All the APIs (`startProject`、`updateProject`、`runGame`、`stopGame`) will return a promise which resolves when the operation finished

Reply:
> As discussed:
> 
> 1. Builder do not need to wait for the previous operation to finish before performing the next operation, the runner will manage that
> 2. All the APIs (`startProject`、`updateProject`、`runGame`、`stopGame`) will return a promise which resolves when the operation finished

https://github.com/goplus/spx/pull/376